### PR TITLE
Add support for Kubernetes 1.27

### PR DESCRIPTION
**Pull request by iblancasa**
Solves #385 and  #365.

* Upgrade KiND to 0.18.0
* Upgrade Kubernetes used versions
* Added support for Kubernetes 1.26
* Added support for Kubernetes 1.27

### CODE REVIEW

File: .github/workflows/e2e.yaml [hunk 1]
```diff
@@ -21,7 +21,7 @@ jobs:
        # should be compatible with them.
        kube-version:
        - "1.21"
-       - "1.25"
+       - "1.27"
```
Comment (frzifus) [inline .github/workflows/e2e.yaml:24]: Shall we do it on 2 steps? That the tests for 1.26 did run at least once? Just for completion :)

Comment (iblancasa) [reply .github/workflows/e2e.yaml:24]: From [this comment](https://github.com/os-observability/tempo-operator/issues/385#issuecomment-1508041834), I supposed is not needed. Also, our thinking is that: if it works with the older and newer versions... we assume it works with the versions between.

File: kind-1.24.yaml [hunk 2]
```diff
@@ -2,4 +2,18 @@ kind: Cluster
 apiVersion: kind.x-k8s.io/v1alpha4
 nodes:
   - role: control-plane
-    image: kindest/node:v1.24.4@sha256:adfaebada924a26c2c9308edd53c6e33b3d4e453782c0063dc0028bdebaddf98
+    image: kindest/node:v1.24.12@sha256:1e12918b8bc3d4253bc08f640a231bb0d3b2c5a9b28aa3f2ca1aee93e1e8db16
+  kubeadmConfigPatches:
```
Comment (andreasgerstmayr) [inline kind-1.24.yaml:6]: Are these new config options required because of the upgrade from `v1.24.4` to `v1.24.12`?

Comment (iblancasa) [reply kind-1.24.yaml:6]: It seems we forget to add those options to 1.24. They should be there from the beginning

**Review by andreasgerstmayr — APPROVED** [review]

**Pull request merged**